frontend/src/components/PercentageBar.tsx

```diff
@@ -0,0 +1,62 @@
+import { FC, forwardRef } from 'react';
+import classNames from 'classnames';
+import colors from '../colors.module.scss';
+import css from './PercentageBar.module.scss';
+
+const classes = classNames.bind(css);
+
+const BarSection: FC<{
+  size: number;
+  color: string;
+  vertical?: boolean;
+  width: number;
+}> = ({ size, color, vertical, width, children }) => (
+  <div
+    style={{
+      zIndex: 10,
+      backgroundColor: color,
+      userSelect: 'none',
+      height: vertical ? `${100 * size}%` : width,
+      width: vertical ? width : `${100 * size}%`,
+    }}
+  >
+    {children}
+  </div>
+);
+
+export const PercentageBar: FC<{
+  stakes: { value: number; id: number | string; color: string }[];
+  totalValue: number;
+  vertical?: boolean;
+  width: number;
+}> = forwardRef(
+  ({ stakes, totalValue, vertical, width, ...props }, ref: any) => (
+    <div
+      {...{ ...props, ref /* for mui tooltip */ }}
+      className={classes(css.stakes, {
+        [css.vertical]: vertical,
+      })}
+    >
+      {totalValue > 0 ? (
+        stakes
+          .filter(({ value }) => value > 0)
+          .map((entry) => (
+            <BarSection
+              key={entry.id}
+              size={entry.value / totalValue}
+              color={entry.color}
+              vertical={vertical}
+              width={width}
+            />
```

Comment:
An option to show the percentages in the bar could be added here. Though it requires a bit of "prop drilling" for showPercentages
```suggestion
            >
              {showPercentages && percent(0).format(entry.value / totalValue)}
            </BarSection>
```

Reply:
I preferred not to add it yet since I dont have a solution to the problem where the section is too small to have a readable percentage

Reply:
I see - that's understandable